### PR TITLE
Implement endpoint for retrieving patients by doctor id

### PR DESCRIPTION
**Pull request by SChandrakanth15**
- Added a new GET endpoint in the DoctorPatientAssignmentController to fetch patients assigned to a specific doctor.
- Created a method in the DoctorPatientAssignmentController to handle requests for patient data based on the provided doctor id
- Integrated the endpoint with the service layer to fetch and return patient details associated with the given doctor.
- Added appropriate validations to manage cases where the doctor id is invalid, or no patients are found.

### CODE REVIEW

File: src/main/java/com/theelixrlabs/healthcare/controller/DoctorPatientAssignmentController.java [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+import com.theelixrlabs.healthcare.dto.DoctorWithPatientsDto;
 import com.theelixrlabs.healthcare.response.SuccessResponse;
 import com.theelixrlabs.healthcare.service.DoctorPatientAssignmentService;
 import jakarta.validation.Valid;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
Comment (vinodkadam22) [inline src/main/java/com/theelixrlabs/healthcare/controller/DoctorPatientAssignmentController.java]: Remove wildcard imports

File: src/main/java/com/theelixrlabs/healthcare/constants/DoctorPatientAssignmentConstants.java [hunk 2]
```diff
@@ -8,6 +8,7 @@ public class DoctorPatientAssignmentConstants {
     public static final String DOCTOR_ALREADY_ASSIGNED_KEY = "doctor.already.assigned";
     public static final String DOCTOR_ALREADY_UNASSIGNED_KEY = "doctor.already.unassigned";
     public static final String DOCTOR_ID_INVALID_KEY = "doctor.id.invalid";
+    public static final String DOCTOR_ID_PARAM = "doctorId";
     public static final String DOCTOR_NOT_FOUND_KEY = "doctor.id.not.found";
     public static final String NO_ASSIGNMENT_EXISTS_KEY = "assignment.not.exists";
     public static final String PATIENT_ID_INVALID_KEY = "patient.id.invalid";
```
Comment (vinodkadam22) [inline src/main/java/com/theelixrlabs/healthcare/constants/DoctorPatientAssignmentConstants.java]: Rename PATIENT_NOT_FOUND_KEY  to PATIENT_ID_NOT_FOUND_KEY

**Review by vinodkadam22 — CHANGES_REQUESTED** [review]

File: src/main/java/com/theelixrlabs/healthcare/controller/DoctorPatientAssignmentController.java (hunk 1 above)
Comment (SChandrakanth15) [reply src/main/java/com/theelixrlabs/healthcare/controller/DoctorPatientAssignmentController.java]: removed the wildcard imports.

File: src/main/java/com/theelixrlabs/healthcare/constants/DoctorPatientAssignmentConstants.java (hunk 2 above)
Comment (SChandrakanth15) [reply src/main/java/com/theelixrlabs/healthcare/constants/DoctorPatientAssignmentConstants.java]: Renamed PATIENT_NOT_FOUND_KEY to PATIENT_ID_NOT_FOUND_KEY.

File: src/main/java/com/theelixrlabs/healthcare/constants/DoctorPatientAssignmentConstants.java [hunk 3]
```diff
@@ -8,6 +8,7 @@ public class DoctorPatientAssignmentConstants {
     public static final String DOCTOR_ALREADY_ASSIGNED_KEY = "doctor.already.assigned";
     public static final String DOCTOR_ALREADY_UNASSIGNED_KEY = "doctor.already.unassigned";
     public static final String DOCTOR_ID_INVALID_KEY = "doctor.id.invalid";
```
Comment (EXR-85) [inline src/main/java/com/theelixrlabs/healthcare/constants/DoctorPatientAssignmentConstants.java:10]: Rename DOCTOR_ID_INVALID_KEY to INVALID_ DOCTOR_ID_KEY

File: src/main/java/com/theelixrlabs/healthcare/constants/DoctorPatientAssignmentConstants.java (hunk 2 above)
Comment (EXR-85) [inline src/main/java/com/theelixrlabs/healthcare/constants/DoctorPatientAssignmentConstants.java:14]: Rename PATIENT_ID_INVALID_KEY to INVALID_PATIENT_ID_KEY

File: src/main/java/com/theelixrlabs/healthcare/constants/DoctorPatientAssignmentConstants.java (hunk 3 above)
Comment (divya-sathyan) [inline src/main/java/com/theelixrlabs/healthcare/constants/DoctorPatientAssignmentConstants.java]: Rename DOCTOR_ID_INVALID_KEY to INVALID_ DOCTOR_ID_KEY

File: src/main/java/com/theelixrlabs/healthcare/constants/DoctorPatientAssignmentConstants.java (hunk 2 above)
Comment (divya-sathyan) [inline src/main/java/com/theelixrlabs/healthcare/constants/DoctorPatientAssignmentConstants.java]: Rename PATIENT_ID_INVALID_KEY to INVALID_PATIENT_ID_KEY

File: src/main/java/com/theelixrlabs/healthcare/dto/DoctorWithPatientsDto.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.experimental.SuperBuilder;
+import java.util.List;
+
+/**
+ * Data Transfer Object (DTO) representing a doctor along with their assigned patients.
+ */
+@Data
+@AllArgsConstructor
+@SuperBuilder
+public class DoctorWithPatientsDto {
```
Comment (ashutosh-2609) [inline src/main/java/com/theelixrlabs/healthcare/dto/DoctorWithPatientsDto.java]: what about renaming DoctorWithPatientsDto to DoctorWithAssignedPatientsDto ?

File: src/main/java/com/theelixrlabs/healthcare/dto/DoctorWithPatientsDto.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    private DoctorDto doctor;
+    private List<PatientDto> patients;
```
Comment (ashutosh-2609) [inline src/main/java/com/theelixrlabs/healthcare/dto/DoctorWithPatientsDto.java]: patients --> assignedPatientsList

File: src/main/java/com/theelixrlabs/healthcare/service/DoctorService.java [hunk 6]
```diff
@@ -42,8 +42,8 @@ public DoctorService(DoctorRepository doctorRepository, DoctorPatientAssignmentR
      * @return true if the doctor is assigned to at least one patient, false otherwise.
      */
     private boolean isDoctorAssignedToPatient(UUID validDoctorId) {
-        Optional<DoctorPatientAssignmentModel> doctorAssignmentList = doctorPatientAssignmentRepository.findByDoctorIdAndDateOfUnassignmentNull(validDoctorId);
-        return doctorAssignmentList.isPresent();
+        List<DoctorPatientAssignmentModel> doctorAssignmentList = doctorPatientAssignmentRepository.findByDoctorIdAndDateOfUnassignmentNull(validDoctorId);
+        return !doctorAssignmentList.isEmpty();
```
Comment (ashutosh-2609) [inline src/main/java/com/theelixrlabs/healthcare/service/DoctorService.java]: can you try using existsByDoctorIdAndDateOfUnassignmentNull(validDoctorId) ?

File: src/main/resources/messages.properties [hunk 7]
```diff
@@ -6,13 +6,14 @@ doctor.aadhaar.number.mandatory = Aadhaar number is mandatory
...
 doctor.already.unassigned = The doctor has already been unassigned.
-doctor.delete.success.message=Doctor with id : {0} deleted Successfully.
+doctor.delete.success.message = Doctor with id : {0} deleted Successfully.
 doctor.deletion.failed.assigned.to.patient = Deletion Failed. Doctor is currently assigned to a Patient.
 doctor.department.mandatory = Department is mandatory
 doctor.department.should.not.be.empty = Department should not be empty
 doctor.first.name.character = First name should be character only
 doctor.first.name.mandatory = First name is mandatory
 doctor.first.name.should.not.be.empty = First name should not be empty
+doctor.id.cannot.be.empty = The doctor'id cannot be empty. Please provide a valid id.
```
Comment (ashutosh-2609) [inline src/main/resources/messages.properties]:
```suggestion
doctor.id.cannot.be.empty = The doctorId cannot be empty. Please provide a valid id.
```

**Review by ashutosh-2609 — CHANGES_REQUESTED** [review]

File: src/main/java/com/theelixrlabs/healthcare/constants/DoctorPatientAssignmentConstants.java (hunk 3 above)
Comment (SChandrakanth15) [reply src/main/java/com/theelixrlabs/healthcare/constants/DoctorPatientAssignmentConstants.java]: Renamed DOCTOR_ID_INVALID_KEY to INVALID_ DOCTOR_ID_KEY.

File: src/main/java/com/theelixrlabs/healthcare/constants/DoctorPatientAssignmentConstants.java (hunk 2 above)
Comment (SChandrakanth15) [reply src/main/java/com/theelixrlabs/healthcare/constants/DoctorPatientAssignmentConstants.java]: Renamed PATIENT_ID_INVALID_KEY to INVALID_PATIENT_ID_KEY.

File: src/main/java/com/theelixrlabs/healthcare/dto/DoctorWithPatientsDto.java (hunk 4 above)
Comment (SChandrakanth15) [reply src/main/java/com/theelixrlabs/healthcare/dto/DoctorWithPatientsDto.java]: Renamed the class DoctorWithPatientsDto  to DoctorWithAssignedPatientsDto

File: src/main/java/com/theelixrlabs/healthcare/dto/DoctorWithPatientsDto.java (hunk 5 above)
Comment (SChandrakanth15) [reply src/main/java/com/theelixrlabs/healthcare/dto/DoctorWithPatientsDto.java]: renamed patients to assignedPatientsList.

File: src/main/java/com/theelixrlabs/healthcare/service/DoctorService.java (hunk 6 above)
Comment (SChandrakanth15) [reply src/main/java/com/theelixrlabs/healthcare/service/DoctorService.java]: changed to existsByDoctorIdAndDateOfUnassignmentNull(validDoctorId).

File: src/main/resources/messages.properties (hunk 7 above)
Comment (SChandrakanth15) [reply src/main/resources/messages.properties]: renamed "doctor' id" to doctorId..

**Review by ashutosh-2609 — APPROVED** [review]

**Review by divya-sathyan — APPROVED** [review]

**Review by vinodkadam22 — APPROVED** [review]

**Pull request merged**